src/server_manager/web_app/app.ts

```diff
@@ -624,32 +617,26 @@ export class App {
   //       return the UI to its exact prior state. Fortunately, the most likely
   //       time to discover an invalid access token is when the application
   //       starts.
-  private digitalOceanRetry = <T>(f: () => Promise<T>): Promise<T> => {
-    return f().catch((e) => {
-      if (!(e instanceof digitalocean_api.XhrError)) {
-        return Promise.reject(e);
-      }
+  digitalOceanRetry<T>(fn: () => Promise<T>): Promise<T> {
```

Comment:
What happened to the idea of emitting an event from  the repo instead?

This flow is a bit convoluted.

Reply:
Thoughts on whether this should be an app event or domain event?